### ISSUE TITLE
Concept Drift Stream Optimisation

### ISSUE THREAD

**Issue by MelchiorChui** [scikit-multiflow/scikit-multiflow#52]
### Current Implementation
```python
    def next_sample(self, batch_size=1):
        self.current_sample_x = None
        self.current_sample_y = None

        for j in range(batch_size):
            self.sample_idx += 1
            x = -4.0 * float(self.sample_idx - self.position) / float(self.width)
            probability_drift = 1.0 / (1.0 + np.exp(x))
            if self.random_state.rand() > probability_drift:
                if self.current_sample_x is None:
                    self.current_sample_x, self.current_sample_y = self._input_stream.next_sample()
                else:
                    X, y = self._input_stream.next_sample()
                    self.current_sample_x = np.append(self.current_sample_x, X, axis=0)
                    self.current_sample_y = np.append(self.current_sample_y, y, axis=0)
            else:
                if self.current_sample_x is None:
                    self.current_sample_x, self.current_sample_y = self._drift_stream.next_sample()
                else:
                    X, y = self._drift_stream.next_sample()
                    self.current_sample_x = np.append(self.current_sample_x, X, axis=0)
                    self.current_sample_y = np.append(self.current_sample_y, y, axis=0)

        return self.current_sample_x, self.current_sample_y
```

### Suggested Implementation
```python
    def next_sample(self, batch_size=1):
        self.current_sample_x = np.zeros((batch_size, self.n_features))
        self.current_sample_y = np.zeros((batch_size, self.n_targets))

        for j in range(batch_size):
            self.sample_idx += 1
            x = -4.0 * float(self.sample_idx - self.position) / float(self.width)
            probability_drift = 1.0 / (1.0 + np.exp(x))
            if self.random_state.rand() > probability_drift:
                X, y = self._input_stream.next_sample()
            else:
                X, y = self._drift_stream.next_sample()
            self.current_sample_x[j, :] = X
            self.current_sample_y[j, :] = y

        return self.current_sample_x, self.current_sample_y.flatten()
```

When nesting multiple ConceptDriftStream instances and computing 100000 instances, next_sample would bottleneck the code due to repeated calls to np.append.

By initialising current_sample_x and current_sample_y and assigning values instead, we were able to achieve a speed up of over 20x.

**Comment by jacobmontiel** [comment]
Excellent suggestion @MelchiorChui! Nice workaround for the append operation. Can you create a pull request for this change?

**Comment by MelchiorChui** [comment]
Done, thank you! :)

**Comment by jacobmontiel** [comment]
Implemented in #53
